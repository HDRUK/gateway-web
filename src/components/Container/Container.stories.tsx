import type { Meta, StoryObj } from "@storybook/react";
import Box from "../Box";
import Container from "./Container";

/**
 * Documents:
 * https://mui.com/system/react-container/
 */

const meta: Meta<typeof Container> = {
    component: Container,
    title: "Layout/Container",
};

export default meta;

type Story = StoryObj<typeof Container>;

export const MaxWidth: Story = {
    render: () => (
        <Container>
            <Box>Max width container</Box>
        </Container>
    ),
};
