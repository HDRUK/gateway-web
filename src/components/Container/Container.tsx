import MuiContainer, {
    ContainerProps as MuiContainerProps,
} from "@mui/material/Container";

const Container = (props: MuiContainerProps) => {
    const { maxWidth = "desktop", ...rest } = props;
    return <MuiContainer maxWidth={maxWidth} {...rest} />;
};

export default Container;
