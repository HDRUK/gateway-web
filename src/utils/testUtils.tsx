import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
