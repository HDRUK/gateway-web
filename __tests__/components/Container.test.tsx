import React from "react";
import Container from "@/components/Container";
import { render } from "../testUtils";

describe("Container", () => {
    it("should render component", async () => {
        const wrapper = render(<Container>Container content</Container>);

        expect(wrapper.container).toMatchSnapshot();
    });
});
