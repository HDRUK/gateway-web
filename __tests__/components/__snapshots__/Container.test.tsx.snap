// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render component 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthDesktop mui-style-1mqm4hl-MuiContainer-root"
  >
    Container content
  </div>
</div>
`;
